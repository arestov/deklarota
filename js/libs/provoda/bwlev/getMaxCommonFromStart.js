export const getMaxCommonFromStart = (arrA, arrB) => {
  const length = Math.min(arrA.length, arrB.length)
  for (let i = 0; i < length; i++) {
    const curA = arrA[i]
    const curB = arrB[i]
    if (curA !== curB) {
      return i
    }
  }

  return length
}
