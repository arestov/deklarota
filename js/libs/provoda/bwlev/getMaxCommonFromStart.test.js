import {it, expect} from '@jest/globals'

import { getMaxCommonFromStart, getMaxCommonFromEnd } from './getMaxCommonFromStart'

it('getMaxCommonFromStart', () => {
  expect(getMaxCommonFromStart([1, 2, 3], [1, 2, 3])).toMatchSnapshot()
  expect(getMaxCommonFromStart([1, 77, 89], [1, 2, 3])).toMatchSnapshot()
  expect(getMaxCommonFromStart([9, 77, 89], [1, 2, 3])).toMatchSnapshot()
  expect(getMaxCommonFromStart([1, 2, 3], [1, 2, 4])).toMatchSnapshot()

  expect(getMaxCommonFromStart([1, 2, 3, 4, 5, 6], [1, 2, 3])).toMatchSnapshot()

})

it('getMaxCommonFromEnd', () => {
  expect(getMaxCommonFromEnd([1, 2, 3], [1, 2, 3])).toMatchSnapshot()
  expect(getMaxCommonFromEnd([4, 77, 3], [1, 2, 3])).toMatchSnapshot()
  expect(getMaxCommonFromEnd([9, 77, 89], [1, 2, 3])).toMatchSnapshot()
  expect(getMaxCommonFromEnd([4, 2, 3], [1, 2, 3])).toMatchSnapshot()

  expect(getMaxCommonFromEnd([6, 5, 4, 1, 2, 3], [1, 2, 3])).toMatchSnapshot()

  expect(getMaxCommonFromEnd([1, 2, 3], [1, 5, 3])).toMatchSnapshot()
})
