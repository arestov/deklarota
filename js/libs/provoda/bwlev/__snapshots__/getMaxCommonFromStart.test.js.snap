// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMaxCommonFromEnd 1`] = `3`;

exports[`getMaxCommonFromEnd 2`] = `1`;

exports[`getMaxCommonFromEnd 3`] = `0`;

exports[`getMaxCommonFromEnd 4`] = `2`;

exports[`getMaxCommonFromEnd 5`] = `3`;

exports[`getMaxCommonFromEnd 6`] = `1`;

exports[`getMaxCommonFromStart 1`] = `3`;

exports[`getMaxCommonFromStart 2`] = `1`;

exports[`getMaxCommonFromStart 3`] = `0`;

exports[`getMaxCommonFromStart 4`] = `2`;

exports[`getMaxCommonFromStart 5`] = `3`;
